File: src/main/java/com/wanderers/hotelier_webservices/rest/AbstractApiDelegate.java
package com.wanderers.hotelier_webservices.rest;

import org.springframework.beans.factory.annotation.Autowired;

import javax.servlet.http.HttpServletRequest;
import java.util.Optional;

/**
 * Class is responsible for encapsulating common attributes of request delegates.
 */
public abstract class AbstractApiDelegate {

    private final HttpServletRequest servletRequest;

    @Autowired
    AbstractApiDelegate(HttpServletRequest servletRequest) {
        this.servletRequest = servletRequest;
    }

    protected HttpServletRequest getServletRequest() {
        return Optional.ofNullable(servletRequest).orElseThrow(() -> new IllegalStateException("request is required"));
    }

}

File: src/main/kotlin/com/wanderers/hotelier_webservices/rest/delegate/AbstractApiDelegate.kt
package com.wanderers.hotelier_webservices.rest.delegate

import org.springframework.beans.factory.annotation.Autowired
import java.util.*
import javax.servlet.http.HttpServletRequest

/**
 * Class is responsible for encapsulating common attributes of request delegates.
 */
abstract class AbstractApiDelegate @Autowired internal constructor(private val servletRequest: HttpServletRequest) {
    protected fun getServletRequest(): HttpServletRequest {
        return Optional.ofNullable(servletRequest).orElseThrow {
            IllegalStateException(
                "request object is required"
            )
        }
    }
}